### DISCUSSION TITLE
Unnecessary tasks are being executed for optional tasks

### DISCUSSION THREAD

**Discussion by cjpearson** [vercel/turborepo#9565] [Help]
### Summary

I'm not sure if this is a misconfiguration, or a bug with turbo. The general setup is I have a repo with 14 packages, each with a "foo" script. 

When I run `turbo foo`, I get `Tasks: 14 successful, 14 total`

Two packages have an additional "bar" script. When bar has no `dependsOn`, I can run `turbo bar` and get `Tasks: 2 successful, 2 total`

But when "bar" depends "foo" being run in the same package, running `turbo bar` results in `Tasks: 16 successful, 16 total`. 

It runs `foo` in all 14 packages rather than the two packages that is necessary. Is it possible for turbo to only run `foo` for the two packages which need to run `bar`?

**Comment by anthonyshew** [comment]
It sounds to me like the behavior of the graph is correct, but possible that the amount of tasks being reported isn't. It's a little hard to say without a reproduction. Would you be able to open an Issue with a reproduction so we can take a look?

**Reply by cjpearson** [reply]
Sure thing, I created a much smaller reproduction but it still seems to have the same issue. https://github.com/vercel/turborepo/issues/9589
